### PR TITLE
chore: golang calculator and httpstan helm charts

### PR DESCRIPTION
**Pull request by Ubisoft-potato**
## What I am doing?
- This pull request is part of imlementation of this https://github.com/bucketeer-io/bucketeer/issues/356.
- This PR will add a new Helm Chart that contains 3 containers:
  -  Golang calculator service
  -  [HttpStan](https://github.com/bucketeer-io/bucketeer/pull/379)
  -  Envoy
- The Golang calculator service will directly request to the HttpStan container by using `localhost` in the pod

### Note
Golang calculator service and HttpStan will be CPU sensitive services, so we may need ajust the resource of the pod according to real load.

### CODE REVIEW

File: manifests/bucketeer/charts/experiment-calculator/templates/envoy-configmap.yaml [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+          load_assignment:
+            cluster_name: environment
+            endpoints:
+              - lb_endpoints:
+                  - endpoint:
+                      address:
+                        socket_address:
+                          address: environment.{{ .Values.namespace }}.svc.cluster.local
+                          port_value: 9000
+          name: environment
```
Comment (masaaania) [inline manifests/bucketeer/charts/experiment-calculator/templates/envoy-configmap.yaml:78]: nit:
Please move the `name` field to the top(L:65) for consistency and visibility.

File: manifests/bucketeer/charts/experiment-calculator/values.yaml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  type: ClusterIP
+  clusterIP: None
+  externalPort: 9000
+
+health:
+  periodSeconds: 10
+  failureThreshold: 10
+  # It is necessary to wait for the model compilation to be done.
+  # The duration is up to resources.
+  # cf. CPU: 500m, MEM: 3Gi -> 280sec
```
Comment (masaaania) [inline manifests/bucketeer/charts/experiment-calculator/values.yaml:82]: 👏

**Review by masaaania — COMMENTED** [review]
Great!

File: manifests/bucketeer/charts/experiment-calculator/templates/envoy-configmap.yaml [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+          health_checks:
+            - grpc_health_check: {}
+              healthy_threshold: 1
+              interval: 10s
+              interval_jitter: 1s
+              no_traffic_interval: 2s
+              timeout: 1s
+              unhealthy_threshold: 2
+          ignore_health_on_host_removal: true
+        - name: environment
```
[the remark this answers is not in the document]
Comment (Ubisoft-potato) [reply manifests/bucketeer/charts/experiment-calculator/templates/envoy-configmap.yaml:65]: OK, I'll update this to `backend` service.

[the remark this answers is not in the document]
Comment (Ubisoft-potato) [reply manifests/bucketeer/charts/experiment-calculator/templates/envoy-configmap.yaml]: @cre8ivejp Envoy configuration file updated.

Also add timezone env to `deployment.yaml`.
Reference:
https://github.com/bucketeer-io/bucketeer/pull/395#discussion_r1243378409

[the remark this answers is not in the document]
Comment (cre8ivejp) [reply manifests/bucketeer/charts/experiment-calculator/templates/envoy-configmap.yaml]: @Ubisoft-potato looks good to me! Thank you!

@masaaania, can you review the last 2 commits when you have time?
- [feat: add timezone support](https://github.com/bucketeer-io/bucketeer/pull/420/commits/323fdbef2c427a35aa5ef19382dd864915619beb)
- [feat: use backend monolithic service](https://github.com/bucketeer-io/bucketeer/pull/420/commits/b2784deca4e3788e9178c5b2650b5fc75a0dc34c)

[the remark this answers is not in the document]
Comment (masaaania) [reply manifests/bucketeer/charts/experiment-calculator/templates/envoy-configmap.yaml]: Excellent!
LGTM!

**Review by cre8ivejp — APPROVED** [review]
Thank you!
I'll merge it tomorrow.

File: manifests/bucketeer/charts/experiment-calculator/values.yaml [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    tag: 0.0.1
+    pullPolicy: IfNotPresent
+  config:
+  port: 8080
+  healthPath: v1/health
+  resources: {}
+
+envoy:
+  image:
+    repository: envoyproxy/envoy-alpine
```
Comment (cre8ivejp) [inline manifests/bucketeer/charts/experiment-calculator/values.yaml:55]: Please update the envoy repository and tag version.
https://github.com/bucketeer-io/bucketeer/blob/main/manifests/bucketeer/charts/account-apikey-cacher/values.yaml#L53-L54

File: manifests/bucketeer/charts/experiment-calculator/templates/envoy-configmap.yaml [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+                  typed_config:
+                    '@type': type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
+                    access_log:
+                      name: envoy.access_loggers.file
+                      typed_config:
+                        '@type': type.googleapis.com/envoy.extensions.access_loggers.file.v3.FileAccessLog
+                        path: /dev/stdout
+                    codec_type: auto
+                    http_filters:
+                      - name: envoy.filters.http.router
```
Comment (cre8ivejp) [inline manifests/bucketeer/charts/experiment-calculator/templates/envoy-configmap.yaml:178]: Please add the `typed_config`.

https://github.com/bucketeer-io/bucketeer/pull/459/files#diff-6c6dfb590e04fdd74ea9cf9c5e456da2a4cea085da6a192c92dc0e67cf6ed317R181-R182

File: manifests/bucketeer/charts/experiment-calculator/templates/envoy-configmap.yaml [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+                          '@type': type.googleapis.com/envoy.extensions.filters.http.health_check.v3.HealthCheck
+                          cluster_min_healthy_percentages:
+                            calculator:
+                              value: 25
+                          headers:
+                            - name: :path
+                              string_match:
+                                exact: /health
+                          pass_through_mode: false
+                      - name: envoy.filters.http.router
```
Comment (cre8ivejp) [inline manifests/bucketeer/charts/experiment-calculator/templates/envoy-configmap.yaml:127]: Please add the `typed_config`.

https://github.com/bucketeer-io/bucketeer/pull/459/files#diff-6c6dfb590e04fdd74ea9cf9c5e456da2a4cea085da6a192c92dc0e67cf6ed317R181-R182

File: manifests/bucketeer/charts/experiment-calculator/templates/deployment.yaml [hunk 7]
```diff
@@ -69,12 +69,6 @@ spec:
           image: "{{ .Values.image.repository }}:{{ .Values.global.image.tag }}"
           imagePullPolicy: {{ .Values.image.pullPolicy }}
           args: ["server"]
-          lifecycle:
-            postStart:
-              httpGet:
-                port: {{ .Values.env.stanPort }}
-                path: {{ .Values.httpstan.healthPath }}
-                scheme: HTTP
```
Comment (Ubisoft-potato) [inline manifests/bucketeer/charts/experiment-calculator/templates/deployment.yaml]: Remove experiment calculator container lifecycle check

**Pull request merged**